pkg/controller/replicaset/replica_set.go

```diff
@@ -198,6 +203,17 @@ func (rsc *ReplicaSetController) Run(workers int, stopCh <-chan struct{}) {
 // getPodReplicaSet returns the replica set managing the given pod.
 // TODO: Surface that we are ignoring multiple replica sets for a single pod.
 func (rsc *ReplicaSetController) getPodReplicaSet(pod *api.Pod) *extensions.ReplicaSet {
+	// look up in the cache, if cached and the cache is valid, just return cached value
+	// Note: if we create a new RS that is also matching a given pod, it will just be ignored
+	// (assuming that the previously matching RS didn't changed), but we sort overlapping RSs
+	// by creation timestamp, so it's no problem. If we change the sort order, we may need
+	// change the code here too.
```

Comment:
You point out a good problem here. We think should add a dirty set, when a RS updated, we put in into the dirty set, add "isCachVaild()" will return false if the RS is in the dirty set.